### ISSUE TITLE
feat: support multiple lock files

### ISSUE THREAD

**Issue by Ayudh** [Ayudh/lock-file-notifier#5]
Some projects might contain more than one lock file
```
project
|__node_modules
|__package.json
|__yarn.lock
|__sub-project-1
|____node_modules
|____package.json
|____yarn.lock
|__sub-project-2
|____node_modules
|____package.json
|____yarn.lock
```
Take lock files as extension setting input and validate accordingly.

**Comment by zardoy** [comment]
Also it doesn't support pnpm, which is finally popular in our days.

@Ayudh I know it has been created a few years ago, but I finally managed to do first-class support for this: https://github.com/zardoy/npm-the-fastest#lockfile-watcher

If you still care of this extension or its users, you or I can backport this solution to here: https://github.com/zardoy/npm-the-fastest/blob/13ae791d8f832b1ca172768bc43ced9497273229/src/autoInstall.ts#L16

So you can just use brace expansion (I don't really remember how this thing in globs is called)  e.g. `**/{yarn.lock,smth.lock}` No setting needed! And this is important! What if switched to branch that use old npm?

The mine the solution overall to install packages in extension might be a bit complicated, but its worth it. The panel is hidden, but revealed when errors, also you can reveal whenever you want it.

Would be happy to hear you!

**Comment by Ayudh** [comment]
Thanks @zardoy for bringing this up. 
It would be great if you can bring your approach here. At the moment, I am occupied but I will definitely look to have this here.

**Comment by zardoy** [comment]
Wow! Thank you so much for that quick response! Yes, please, keep me up with updates and let me know what you think when you have enough time to look at it as I'm really interested in this.

Looking forward for any kind of questions! 😄

**Comment by Ayudh** [comment]
Sorry for the delay.
It would be great if you can bring the pnpm support and other features you mentioned here.
But for autoinstall without prompt, I would still like to have a prompt before install because most of the times, I would just switch to a branch just for a brief moment and come back to the original.
Anyway, I see extension setting for the autoinstall (correct me if I am wrong).

Please provide a list of changes that you would like to see and corresponding links to the sample implementation in your repo. We can discuss at individual feat level.

Thanks @zardoy

**Comment by zardoy** [comment]
@Ayudh Sorry for the delay, had extremely busy sat.

> But for autoinstall without prompt, I would still like to have a prompt before install because most of the times, I would just switch to a branch just for a brief moment and come back to the original.

Absoletely didn't get the context. Have you checked the source or extension itself? Yes, you can do it by changing the setting, but default behavior is showing a prompt before install.

> Anyway, I see extension setting for the autoinstall (correct me if I am wrong).

You'd better link things as I don't really understand what extension setting for the autoinstall you mean 😃

Anyway, there is a Demo:

![Nov-28-2022 00-10-36](https://user-images.githubusercontent.com/46503702/204159887-c9aaa212-885f-4fdc-bf63-1f01cecf6c4e.gif)

However I'm currently switching default behavior to show prompt only after actual git checkout, to avoid annoyance e.g. after running `yarn` in terminal, so feel free to grab latest changes. Though the prompt might be displayed with some delay (for some reason only Windows has this issue, but there is 2600ms timeout).

> Please provide a list of changes that you would like to see and corresponding links to the sample implementation in your repo. We can discuss at individual feat level.

I think I've already done this above? https://github.com/Ayudh/lock-file-notifier/issues/5#issuecomment-1269386070

But there is updated link with *current* implementation: https://github.com/zardoy/npm-the-fastest/blob/main/src/autoInstall.ts#L11-L124

**Comment by zardoy** [comment]
Also the difference with my implementation is that I don't have a validation feature. I think it'd be probably cool to support it instead of restricting check to git checkouts (package-lock.json: common thing as that only `version` changes). However:
1. `pnpm` (which I mainly use) doesn't support --dry-run and that's why I didn't implement it at all.
2. Doesn't seem to work with yarn (still get notified after `yarn` cli commands)
